pkgs/top-level/all-packages.nix

```
@@ -12372,6 +12372,7 @@ with pkgs;
   sbcl_2_0_9 = callPackage ../development/compilers/sbcl/2.0.9.nix {};
   sbcl_2_1_1 = callPackage ../development/compilers/sbcl/2.1.1.nix {};
   sbcl_2_1_2 = callPackage ../development/compilers/sbcl/2.1.2.nix {};
+  sbcl_2_1_8 = callPackage ../development/compilers/sbcl/2.1.8.nix {};
   sbcl = sbcl_2_1_2;
```

I think you also need to set it as the default sbcl:
```suggestion
  sbcl = sbcl_2_1_8;
```

---

Haven't done any sort of testing on `lispPackages` with that sbcl version. As far as I know SBCL 2.1.8 changes some warnings (likely the versions before as well) which tends to break compilation for some packages.

I've done tests with out of tree packages though and those seemed to cause no issues.